### PR TITLE
[graphql]suins lookup

### PR DESCRIPTION
**Pull request by wlmyng**
## Description 

implement resolveNameServiceAddress on top-level Query resolver + nameServiceConnection for objectOwner.

Not keen on importing all of sui-json-rpc just to get access to the NameService type. Made a copy in sui-graphql-rpc, but thoughts on doing something like lifting the type to sui-json-rpc-types?
Screenshot 2023-10-10 at 3 47 22 PM

Not clear on what defaultNameServiceName should be ... is this intended to be the first/ primary name returned by https://docs.sui.io/sui-jsonrpc#suix_resolveNameServiceNames

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by stefan-mysten — APPROVED** [review]

File: crates/sui-graphql-rpc/src/context_data/db_data_provider.rs [hunk 1]
```diff
@@ -490,9 +495,16 @@ impl PgManager {
         last: Option<u64>,
         before: Option<String>,
         filter: Option<ObjectFilter>,
+        name_service_name: bool,
     ) -> Result<Option<(Vec<StoredObject>, bool)>, Error> {
         let mut query = objects::dsl::objects.into_boxed();
 
+        if name_service_name {
+            query = query.filter(
+                objects::dsl::df_object_type.eq("0xd22b24490e0bae52676651b4f56660a5ff8022a2576e0089f79b3c88d44e08f0::name_record::NameRecord".to_string())
```
Comment (bmwill) [inline crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:504]: Why is this hardcoded in this query?`0xd22b24490e0bae52676651b4f56660a5ff8022a2576e0089f79b3c88d44e08f0` If we want this to be usable on other networks (say testnet) we'll need to have the configurable with a default for mainnet

File: crates/sui-graphql-rpc/src/context_data/name_service.rs [hunk 2]
```diff
@@ -0,0 +1,226 @@
+// Copyright (c) Mysten Labs, Inc.
```
Comment (bmwill) [inline crates/sui-graphql-rpc/src/context_data/name_service.rs:1]: Why is this file copied? This code already exists, please use the existing code vs copying it elsewhere.

File: crates/sui-graphql-rpc/src/context_data/db_data_provider.rs (hunk 1 above)
Comment (bmwill) [reply crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:504]: Also it seems like this parameter is unused? why do we want to filter for this?

**Review by bmwill — CHANGES_REQUESTED** [review]

File: crates/sui-graphql-rpc/src/context_data/db_data_provider.rs (hunk 1 above)
Comment (wlmyng) [reply crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:504]: I tried to do something slick w/ the Domain, NameRecord table but I don't think the approach works. Will remove this code as well

File: crates/sui-graphql-rpc/src/context_data/name_service.rs (hunk 2 above)
Comment (wlmyng) [reply crates/sui-graphql-rpc/src/context_data/name_service.rs]: Pulled in sui-json-rpc crate

**Review by bmwill — APPROVED** [review]

**Pull request merged**